pkg/operator/operator_config_test.go

```diff
@@ -0,0 +1,207 @@
+		},
+		{
+			desc: "missing project",
+			existing: &monitoringv1.OperatorConfig{
+				ObjectMeta: defaultObjectMeta,
+				Collection: monitoringv1.CollectionSpec{
+					ExternalLabels: map[string]string{
+						export.KeyLocation: operatorOpts.Location,
+						export.KeyCluster:  operatorOpts.Cluster,
+						"foo":              "bar",
+					},
+				},
+				Rules: monitoringv1.RuleEvaluatorSpec{
+					ExternalLabels: map[string]string{
+						export.KeyLocation: operatorOpts.Location,
+						export.KeyCluster:  operatorOpts.Cluster,
+						"abc":              "xyz",
+					},
+				},
+			},
+			expected: &defaultOperatorConfigWithExtraLabels,
+		},
+		{
+			desc: "override project",
+			existing: &monitoringv1.OperatorConfig{
+				ObjectMeta: defaultObjectMeta,
+				Collection: monitoringv1.CollectionSpec{
+					ExternalLabels: map[string]string{
+						export.KeyProjectID: "project-other",
+						export.KeyLocation:  operatorOpts.Location,
+						export.KeyCluster:   operatorOpts.Cluster,
+						"foo":               "bar",
+					},
+				},
+				Rules: monitoringv1.RuleEvaluatorSpec{
+					ExternalLabels: map[string]string{
+						export.KeyProjectID: "project-other",
+						export.KeyLocation:  operatorOpts.Location,
+						export.KeyCluster:   operatorOpts.Cluster,
+						"abc":               "xyz",
+					},
+				},
+			},
+			expected: &monitoringv1.OperatorConfig{
+				ObjectMeta: defaultObjectMeta,
+				Collection: monitoringv1.CollectionSpec{
+					ExternalLabels: map[string]string{
+						export.KeyProjectID: "project-other",
+						export.KeyLocation:  operatorOpts.Location,
+						export.KeyCluster:   operatorOpts.Cluster,
+						"foo":               "bar",
+					},
+				},
+				Rules: monitoringv1.RuleEvaluatorSpec{
+					ExternalLabels: map[string]string{
+						export.KeyProjectID: "project-other",
+						export.KeyLocation:  operatorOpts.Location,
+						export.KeyCluster:   operatorOpts.Cluster,
+						"abc":               "xyz",
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			clientBuilder := newFakeClientBuilder()
+			if tc.existing != nil {
+				clientBuilder = clientBuilder.WithObjects(tc.existing.DeepCopy())
+			}
+			kubeClient := clientBuilder.Build()
+
+			reconciler := newOperatorConfigReconciler(kubeClient, operatorOpts)
+			operatorConfig, err := reconciler.ensureOperatorConfig(ctx, logger, reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: DefaultPublicNamespace,
+					Name:      NameOperatorConfig,
+				},
+			})
+			if err != nil {
+				t.Fatalf("ensure operator config: %s", err)
+			}
+
+			// Normalize before comparisons.
+			operatorConfig.ResourceVersion = ""
+
+			if diff := cmp.Diff(operatorConfig, tc.expected); diff != "" {
+				t.Fatalf("unexpected operator config: %s", diff)
+			}
+
+			// Make sure the object is updated in case of defaulting.
+			if tc.existing != nil {
+				existingLatest := tc.existing.DeepCopy()
```

Comment:
suggestion: Why overwrite `existingLatest`?

Couldn't you `Get` into an empty `OperatorConfig` and compare that against `tc.expected`?